src/stk500v2.c

```diff
@@ -1223,6 +1245,109 @@ static int stk500v2_initialize(const PROGRAMMER *pgm, const AVRPART *p) {
     // stk600_setup_isp(pgm); [moved to pgm->enable()]
   }
 
+  // Read or write target voltage
+  if (PDATA(pgm)->vtarg_get || PDATA(pgm)->vtarg_set) {
+    // Read current target voltage set value
+    unsigned char vtarg_read;
+    stk500v2_getparm(pgm, PARAM_VTARGET, &vtarg_read);
+    if (PDATA(pgm)->vtarg_get)
+      msg_info("Target voltage value read as %.2fV\n", (vtarg_read / 10.0));
+    // Write target voltage value
+    else {
+      msg_info("Changing target voltage from %.2f to %.2fV\n", (vtarg_read / 10.0), PDATA(pgm)->vtarg_data);
+      if(pgm->set_vtarget(pgm, PDATA(pgm)->vtarg_data) < 0)
+        return -1;
+    }
+  }
+
+  // Read or write analog reference voltage
+  if (PDATA(pgm)->varef_get || PDATA(pgm)->varef_set) {
+    if(PDATA(pgm)->pgmtype == PGMTYPE_STK500) {
+      // STK500: Read current analog reference voltage
+      unsigned char varef_read;
+      stk500v2_getparm(pgm, PARAM_VADJUST, &varef_read);
+      if (PDATA(pgm)->varef_get)
+        msg_info("Analog reference voltage value read as %.2fV\n", (varef_read / 10.0));
+      // STK500: Write analog reference voltage
+      else {
+        msg_info("Changing analog reference voltage from %.2f to %.2fV\n",
+          (varef_read / 10.0), PDATA(pgm)->varef_data);
+        if(pgm->set_varef(pgm, 0, PDATA(pgm)->varef_data) < 0)
+          return -1;
+      }
+    } else if(PDATA(pgm)->pgmtype == PGMTYPE_STK600) {
+      // STK600: Read current target voltage set value
+      unsigned int varef_read;
+      stk500v2_getparm2(pgm, PDATA(pgm)->varef_channel == 0 ? PARAM2_AREF0 : PARAM2_AREF1, &varef_read);
+      if (PDATA(pgm)->varef_get)
+        msg_info("Analog reference channel %d voltage read as %.2fV\n", PDATA(pgm)->varef_channel, (varef_read / 100.0));
+      // STK600: Write target voltage value for channel n
+      else {
+        msg_info("Changing analog reference channel %d voltage from %.2f to %.2fV\n",
+          PDATA(pgm)->varef_channel, (varef_read / 100.0), PDATA(pgm)->varef_data);
+        if(pgm->set_varef(pgm, PDATA(pgm)->varef_channel, PDATA(pgm)->varef_data) < 0)
+          return -1;
+      }
+    }
+  }
+
+  // Read or write clock generator frequency
+  if (PDATA(pgm)->fosc_get || PDATA(pgm)->fosc_set) {
+    if(PDATA(pgm)->pgmtype == PGMTYPE_STK500) {
+      // Read current target voltage set value
+      unsigned char osc_pscale;
+      unsigned char osc_cmatch;
+      const char *unit_get = {"Hz"};
+      double f_get;
```

Comment:
maybe initialise f_get here? It's only set in certain circs but used in slightly other circs.